src/ASBNApp.Frontend/Components/Export/PillSelectorWeek.razor

```diff
@@ -0,0 +1,139 @@
+@* Simple chip selector component for weeks *@
+
+<MudStack Row="true">
+    <MudChipSet Filter="true" Mandatory="true">
+        <MudChip Text="purple" Default="true" Color="Color.Primary" OnClick="@(e => { UpdateFromWeekPicker(0); })">
+            Current Week (Week @_currentWeek)
+        </MudChip>
+
+        <MudChip Text="purple" Color="Color.Primary" OnClick="@(e => { UpdateFromWeekPicker(1); })">
+            Previous Week (Week @(_currentWeek - 1))
+        </MudChip>
+
+        <MudChip Text="purple" Color="Color.Primary" OnClick="OpenDatePickerOverlay">
+            Pick Custom Week (Week @SelectedWeek) &nbsp;
+            <MudIcon Icon="@Icons.Material.Rounded.CalendarMonth" Color="Color.Surface" />
+        </MudChip>
+    </MudChipSet>
+
+    @* Week picker (technically a date picker, since there is no week picker *@
+    <MudOverlay @bind-Visible="isDatePickerOverlayVisible" DarkBackground="true">
+        <MudDatePicker @ref=_picker PickerVariant="PickerVariant.Static" @bind-Date="date" ShowWeekNumbers="true">
+            <PickerActions>
+                <MudButton Class="mr-auto align-self-start" OnClick="@(() => CancelPicker())">Cancel</MudButton>
+                <MudButton Color="Color.Primary" OnClick="@(() => SetPickerToCurrentWeek())">Current week</MudButton>
+            </PickerActions>
+        </MudDatePicker>
+    </MudOverlay>
+</MudStack>
+
+
+@code {
+    // Two way binding parameters
+    [Parameter]
+    public int? SelectedWeek { get; set; }
+    [Parameter]
+    public EventCallback<int?> SelectedWeekChanged { get; set; }
+    [Parameter]
+    public int? SelectedYear { get; set; }
+    [Parameter]
+    public EventCallback<int?> SelectedYearChanged { get; set; }
+    [Parameter]
+    public EventCallback SelectedDateChanged { get; set; }
+
+    // Single place to trigger the callbacks
+    public void TriggerAllCallbacks(int? week, int? year)
+    {
+        SelectedWeekChanged.InvokeAsync(week);
+        SelectedYearChanged.InvokeAsync(year);
+        SelectedDateChanged.InvokeAsync();
```

Comment:
Important: This way of calling async methods doesn't guarantee an execution order. If your logic doesn't depend on it, it's okay, but should be noted in a comment that this is okay